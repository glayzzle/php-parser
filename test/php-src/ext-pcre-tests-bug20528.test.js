// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/pcre/tests/bug20528.phpt
  it("Bug #20528 (preg_split() drops characters (re-opens Bug #15413))", function () {
    expect(parser.parseCode("<?php\n    $data = '(#11/19/2002#)';\n    var_dump(preg_split('/\\b/', $data));\n?>")).toMatchSnapshot();
  });
});
