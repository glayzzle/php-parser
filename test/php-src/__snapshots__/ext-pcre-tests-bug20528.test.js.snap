// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Bug #20528 (preg_split() drops characters (re-opens Bug #15413)) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "data",
        },
        "operator": "=",
        "right": String {
          "isDoubleQuote": false,
          "kind": "string",
          "raw": "'(#11/19/2002#)'",
          "unicode": false,
          "value": "(#11/19/2002#)",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              String {
                "isDoubleQuote": false,
                "kind": "string",
                "raw": "'/\\\\b/'",
                "unicode": false,
                "value": "/\\\\b/",
              },
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "data",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "preg_split",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
